src/solvers/refinement/string_builtin_function.h

```diff
@@ -32,8 +33,29 @@ class string_builtin_functiont
 
   virtual std::string name() const = 0;
 
-protected:
+  /// Add constraints ensuring that the value of result expression of the
+  /// builtin function corresponds to the value of the function call.
+  virtual exprt
+  add_constraints(string_constraint_generatort &constraint_generator) const = 0;
+
+  exprt return_code;
+
+  /// Tells whether the builtin function can be a testing function, that is a
+  /// function that does not return a string, for instance like `equals`,
+  /// `indexOf` or `compare`.
+  virtual bool maybe_testing_function() const
```

Comment:
Why `maybe` and not `is`?

Reply:
For the functions that don't have yet a dedicated builtin_function class we consider that could be testing function although we don't actually know. Ultimately this can be replaced by a `is`